### ISSUE TITLE
3.2.x codegen doesn't work with MS SQL Server

### ISSUE THREAD

**Issue by Neverminder** [slick/slick#1767]
Codegen basically generates an empty `Tables.scala` looking like this:

```
package db
// AUTO-GENERATED Slick data model
/** Stand-alone Slick data model for immediate use */
object Tables extends {
  val profile = slick.jdbc.SQLServerProfile
} with Tables

/** Slick data model trait for extension, choice of backend or usage in the cake pattern. (Make sure to initialize this late.) */
trait Tables {
  val profile: slick.jdbc.JdbcProfile
  import profile.api._
  import slick.model.ForeignKeyAction

  /** DDL for all tables. Call .create to execute. */
  lazy val schema: profile.SchemaDescription = profile.DDL(Nil, Nil)
  @deprecated("Use .schema instead of .ddl", "3.0")
  def ddl = schema
}
```

Version 3.1.1 works just fine, all that was done is version changed on `build.sbt` from 3.1.1 to 3.2.1, tried 3.2.0 as well. Works fine with other databases though (Postgresql for instance).

**Comment by trevorsibanda** [comment]
Hi @neverminder 

Thanks for reporting this, I have been trying to figure out what exactly could be causing this in your case since you shared this in the Gitter channel. 

Code generator tests are run for mssql and other dbms in 3.2 and 3.2.1 which highly suggests the problem is in your config.

The details you have provided however are not enough to recreate or diagnose the problem

**Comment by Neverminder** [comment]
@trevorsibanda 

I haven't changed my config in any way and it works with 3.1.1

This is console output: https://pastebin.com/zVH5wgpn

I'm running it as sbt task:

```
val conf = ConfigFactory.parseFile(new File("conf/application.conf")).resolve()
val user = conf.getString("slick.dbs.default.db.user")
val pswd = conf.getString("slick.dbs.default.db.password")

val db: TaskKey[Unit] = TaskKey[Unit]("db", "Generate slick DSL")
db := {
	val outputDir = "app"
	val url = "jdbc:sqlserver://192.168.1.7:1433;databaseName=DB_NAME"
	val jdbcDriver = "com.microsoft.sqlserver.jdbc.SQLServerDriver"
	val slickDriver = "slick.jdbc.SQLServerProfile"
	val pkg = "db"
	toError(
		(runner in Compile).value.run(
			"slick.codegen.SourceCodeGenerator",
			(dependencyClasspath in Compile).value.files,
			Array(slickDriver, jdbcDriver, url, outputDir, pkg, user, pswd),
			streams.value.log
		)
	)
	val fname = outputDir + "Tables.scala"
	Seq(file(fname))
}
```
Any ideas?

**Comment by Neverminder** [comment]
So I've tried a different SQL Server database, also JTDS open source driver, but no change whatsoever. I've ran out if ideas, everything literally works just fine by simply changing Slick version back to 3.1.1, which means the 3.2.1 codegen version is just broken for SQL Server databases. Also, what's up with all those debug and benchmark statements, they don't appear with a postgres database.

**Comment by Neverminder** [comment]
@trevorsibanda I was analysing the codegen and I think I found the issue. This is version 3.2.1.

Basically the tables are not retrieved and it all goes wrong in `slick/src/main/scala/slick/jdbc/meta/MTable.scala`

When the method 
`def getTables(cat: Option[String], schemaPattern: Option[String], namePattern: Option[String], types: Option[Seq[String]])` 
is called the value of `schemaPattern` for some reason is `Some(userName)` where userName is the user name I use for database connection, so this is totally bizzare. If I hardcode the value of `schemaPattern.orNull` as `null` - everything works just fine.

I also checked version 3.1.1 and `schemaPattern` value is `None` where it gets turned into null in `schemaPattern.orNull`.

Any ideas?

**Comment by bhollosi** [comment]
I confirm that this fix resolves the problem generating code with an ms sql server.

**Comment by hvesalai** [comment]
@smootoo's original explanation of why user is used as schema:

> For SqlServer, schemas and users aren’t necessarily the same thing, but after SqlServer2008, there are system tables that come back from a query for tables that doesn’t filter by schema. The defaultTables method seems to be intended for list all the tables associated with the current connection and so when system tables come back, tests start to fail.

**Comment by hvesalai** [comment]
@bhollosi which version of SQLServer are you using. It seems the fix breaks the functionality for some SQLServer versions.

**Comment by bhollosi** [comment]
The latest tag of this docker image: 
https://hub.docker.com/r/microsoft/mssql-server-linux/

SQL Server 2017

**Comment by hvesalai** [comment]
Ok, so it seems different SQL Server versions require different implementation.

**Comment by Neverminder** [comment]
@hvesalai  The version I tested this fix against was SQL Server 2008 R2 (10.50.6220) if that helps.

**Comment by hvesalai** [comment]
So strange. The tests fail with `sqlserver2012` and `sqlserver2014 ` with the fix #1811

Ths is because, as @smootoo pointed out, without the filtering, also some non-user tables are listed.. Namely, the following in the tests:

```
Table(QualifiedName(trace_xe_action_map,Some(sys),Some(slicktest)),
Table(QualifiedName(trace_xe_event_map,Some(sys),Some(slicktest)),
```

**Comment by hvesalai** [comment]
Fix merged to master
